deluca/envs/lung/_delay_lung.py

```diff
@@ -65,24 +65,31 @@ def dynamics(self, state, action):
         r = (3.0 * self.volume / (4.0 * jnp.pi)) ** (1.0 / 3.0)
         lung_pressure = self.C_lung * (1 - (self.r0 / r) ** 6) / (self.r0 ** 2 * r)
 
-        if self.T < self.delay:
-            pipe_impulse = 0
-            peep = 0
-        else:
-            pipe_impulse = self.control_gain * self.in_history[0]
-            peep = self.out_history[0]
+        pipe_impulse = jax.lax.cond(self.time < self.delay,
+                                    lambda x : 0.0,
+                                    lambda x: self.control_gain * self.in_history[0],
+                                    None)
+        peep = jax.lax.cond(self.time < self.delay,
+                            lambda x : 0.0,
+                            lambda x: self.out_history[0],
+                            None)
 
         pipe_pressure = self.inertia * state["pipe_pressure"] + pipe_impulse
         pressure = jnp.maximum(0, pipe_pressure - lung_pressure)
 
-        if peep:
-            pipe_pressure *= 0.995
+        pipe_pressure = jax.lax.cond(peep is not None,
+                                     lambda x: x * 0.995,
+                                     lambda x: x,
+                                     pipe_pressure)
 
-        return volume, pressure, pipe_pressure
+        return {'volume': volume, 'pressure': pressure, 'pipe_pressure': pipe_pressure}
```

Comment:
Is this change consistent across all the lung environments? Is the plan to change to a dictionary?

Reply:
The person who wrote delay_lung used state as a dictionary throughout the file, but I guess they forgot to change this line? That's why I changed it like that. It's not a dictionary in the other two envs.

Reply:
Ah, makes sense. Let's make them consistent! Dictionary might be a little easier to understand.

Reply:
ok will convert them to dicts